src/pfe/file-watcher/server/src/utils/kubeutil.ts

```diff
@@ -115,14 +122,25 @@ export async function getApplicationContainerInfo(projectInfo: ProjectInfo, oper
         const internalPorts: Array<string> = [];
         const exposedPorts: Array<string> = [];
         const appPorts = projectInfo.appPorts;
-        const resp = await k8sClient.api.v1.namespaces(KUBE_NAMESPACE).services.get({ qs: { labelSelector: releaseLabel } });
+        let resp: any = undefined;
+        let deploymentconfig: string = undefined;
+
+        if (projectInfo.projectType == "odo") {
+            resp = await k8sClient.api.v1.namespaces(KUBE_NAMESPACE).services.get();
+            const res: string[] = releaseLabel.split("=");
+            deploymentconfig = res[1].trim();
+        } else {
+            resp = await k8sClient.api.v1.namespaces(KUBE_NAMESPACE).services.get({ qs: { labelSelector: releaseLabel } });
+        }
 
         // iterate through the available ports and store both the internal and exposed ports
         for ( let i = 0 ; i < resp.body.items.length ; i++ ) {
-            info.serviceName = resp.body.items[i].metadata.name;
-            for (let j = 0; j < resp.body.items[i].spec.ports.length; j++) {
-                internalPorts.push(String(resp.body.items[i].spec.ports[j].targetPort));
-                exposedPorts.push(String(resp.body.items[i].spec.ports[j].nodePort));
+            if (projectInfo.projectType != "odo" || (projectInfo.projectType == "odo" && resp.body.items[i].spec.selector.deploymentconfig == deploymentconfig)) {
```

Comment:
Same here. We should delegate that to the extension code instead of hardcoding on this layer of the code.